Readme.md

```diff
@@ -1,87 +1,153 @@
+You can find MREPL output for a simple "CREATE-INSERT-SELECT" scenario below. Mind the second argument to sqlite3_open_v2() that is an OR-ed SQLite flag values. 6 in this context is `SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE`.
+
+
+```bash
+$cargo install marine mrepl
+...
+$mrepl -q Config.toml
+
+1> call sqlite3 sqlite3_open_v2 [":memory:",6,""]
+result: {
+  "db_handle": 198600,
+  "ret_code": 0
+}
+ elapsed time: 357.556µs
+
+2> call sqlite3 sqlite3_exec [198600, "CREATE TABLE tab1(i bigint);", 0, 0]
+result: {
+  "err_msg": "",
+  "ret_code": 0
+}
+ elapsed time: 1.736661ms
+
+3> call sqlite3 sqlite3_exec [198600, "INSERT INTO tab1 VALUES (42);", 0, 0]
+result: {
+  "err_msg": "",
+  "ret_code": 0
+}
+ elapsed time: 330.098µs
+
+4> call sqlite3 sqlite3_prepare_v2 [198600, "SELECT * FROM tab1;"]
+result: {
+  "ret_code": 0,
+  "stmt_handle": 244584,
+  "tail": 268147
+}
+ elapsed time: 280.668µs
+
+5> call sqlite3 sqlite3_step [244584]
+result: 100
+ elapsed time: 124.122µs
+
+6> call sqlite3 sqlite3_column_int64 [244584,0]
+result: 42
+ elapsed time: 72.483µs
+
 ```
 
-# How to use
-
-At now, this fork exports five API functions:
-```cpp
-
-/**
- * Executes given SQL request and returns result in as a pointer to the following structure: | result size (4 bytes, le)| result (size bytes) |.
- *
- * @param sql a pointer to the supplied sql request
- * @param length a size of the supplied sql request
- * @return a pointer to the struct contains result_size and result
- */
-char *invoke(const char *sql, size_t length);
-
-/**
- * Allocates a memory region of a given size. Could be used by Wasm execution environments for byte array passing.
- *
- * @param size a size of allocated memory region
- * @return a pointer to the allocated memory region
- */
-void *allocate(size_t size);
-
-/**
- * Frees a memory region. Could be used by Wasm execution environments for freeing previous memory allocated by `allocate` function.
- *
- * @param ptr a pointer to the previously allocated memory region
- * @param size a size of the previously allocated memory region
- */
-void deallocate(void *ptr, size_t size);
-
-/**
- * Stores one byte by a given address in the module memory.
- *
- * @param ptr a address where byte should be stored
- * @param value a byte to be stored
- */
-void store(void *ptr, char value);
-
-/**
- * Returns one byte by a given address in the module memory.
- *
- * @param ptr a address at which the needed byte is located
- * @return the byte at the given address
- */
-char load(void *ptr);
+
+
+You can also try the SQLite Wasm module using Rust together with [this Sqlite connector](https://github.com/fluencelabs/sqlite-wasm-connector). [Here](https://github.com/fluencelabs/examples/tree/main/marine-examples/sqlite) you can find a simple SQL REPL utility example built into a Wasm module. Clone the repo, navigate to the marine-examples/sqlite and run build.sh to bootstrap the SQL REPL example.
+
```

Comment:
This contradicts with @mikevoronov intentions so I prefer to left this part.

Reply:
I think that @boneyard93501's approach is better because you already made a cool example with raw SQLite API. So suggest moving this example to examples and putting a link here.

Reply:
Fixed